Src/java/fhir-r4/src/main/java/org/cqframework/cql/cql2elm/fhir/r4/FhirModelInfoProvider.java

```diff
@@ -0,0 +1,53 @@
+package org.cqframework.cql.cql2elm.fhir.r4;
+
+import org.cqframework.cql.cql2elm.ModelInfoProvider;
+import org.cqframework.cql.cql2elm.NamespaceAware;
+import org.cqframework.cql.cql2elm.NamespaceManager;
+import org.hl7.elm.r1.VersionedIdentifier;
+import org.hl7.elm_modelinfo.r1.ModelInfo;
+import org.hl7.elm_modelinfo.r1.serializing.ModelInfoReaderFactory;
+
+import java.io.IOException;
+
+/**
+ * Created by Bryn on 4/15/2016.
+ */
+public class FhirModelInfoProvider implements ModelInfoProvider, NamespaceAware {
+    private NamespaceManager namespaceManager;
+
+    public void setNamespaceManager(NamespaceManager namespaceManager) {
+        this.namespaceManager = namespaceManager;
+    }
+
+    private boolean isFHIRModelIdentifier(VersionedIdentifier modelIdentifier) {
+        if (namespaceManager != null && namespaceManager.hasNamespaces()) {
+            return modelIdentifier.getId().equals("FHIR") &&
+                    (modelIdentifier.getSystem() == null || modelIdentifier.getSystem().equals("http://hl7.org/fhir"));
+        }
+
+        return modelIdentifier.getId().equals("FHIR");
+    }
+
+    public ModelInfo load(VersionedIdentifier modelIdentifier) {
+        if (isFHIRModelIdentifier(modelIdentifier)) {
+            String localVersion = modelIdentifier.getVersion() == null ? "" : modelIdentifier.getVersion();
+            try {
+                switch (localVersion) {
+                    case "4.0.1":
```

Comment:
Are we still working on this? Why does it only load 4.0.1 and not the others?

Reply:
The fhir-all package loads all fhir models, the fhir-r4 one only loads 4.0.1.